File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> p = new PriorityQueue<Integer>();
        for (int i = 0;i<nums.length;i++)
        {
            if (p.size() < k)
            {
                p.add(nums[i]);
            }
            else
            {
                if (nums[i] <= p.peek())
                {
                    
                }
                else
                {
                    p.add(nums[i]);
                    p.remove();
                }
            }
        }
        int a = p.peek();
        return a;
    }
}

File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.py
class Solution(object):
    def findKthLargest(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        if k > len(nums):
            return -1
        max_heap = []  
        for num in nums:
            heapq.heappush(max_heap, -num)
            if len(max_heap) > len(nums) - k + 1:
                heapq.heappop(max_heap)
            # print(max_heap)
        return -max_heap[0]
